scene/gui/licenses_dialog.h

```diff
@@ -0,0 +1,64 @@
+/**************************************************************************/
+/*  licenses_dialog.h                                                     */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef LICENSES_DIALOG_H
+#define LICENSES_DIALOG_H
+
+#include "box_container.h"
+#include "button.h"
+#include "label.h"
+#include "margin_container.h"
+#include "panel_container.h"
+#include "rich_text_label.h"
```

Comment:
This should be forward-declared and includes moved to cpp.

Reply:
I've tried to move the includes to .cpp and add `class CanvasLayer;` to the header, but I get the following:

```
In file included from ./scene/main/scene_tree.cpp:49,
                 from scene/main/scu/scu_scene_main_1.gen.cpp:3:
./scene/gui/licenses_dialog.h:38:31: error: invalid use of incomplete type 'class CanvasLayer'
   38 | class LicensesDialog : public CanvasLayer {
      |                               ^~~~~~~~~~~
In file included from ./scene/main/window.h:34,
                 from ./scene/main/node.cpp:42,
                 from scene/main/scu/scu_scene_main_1.gen.cpp:1:
./scene/main/viewport.h:47:7: note: forward declaration of 'class CanvasLayer'
   47 | class CanvasLayer;
      |       ^~~~~~~~~~~
```

It works if I keep the `#include "scene/main/canvas_layer.h"` include though.